tests/e2e/helpers.go

```diff
@@ -257,33 +241,6 @@ func executeTests(t *testing.T, tests ...testExecution) {
 	defer mcTctx.cleanupTrackedInvalidPools()
 	defer mcTctx.cleanupTrackedPools()
```

Comment:
These are only used in a couple of tests, but once we remove them here, this method won't contain any setup or teardown code. That will allows us to break the tests into separate suites for serial and parallel tests, and remove the custom runner.

Reply:
After doing some more digging, we discovered these are lazy loaded so they don't reboot machines in the pool when they're created.

After testing with 4.12, those machine config pools don't cause reboots when they're created. We can either create them in setup, regardless of their use, or we can create them in the tests still.